### PR TITLE
TR-72: Add TS and tests to releases derive

### PR DESCRIPTION
**Pull request by shoumaw**
https://bitovi.atlassian.net/browse/TR-72
Added type safety to

- releases/derive
- releases/semver-releases (not being used, but migrated it anyways for consistency)
- releases/unique-trailing-names

migrate releases/derive test from js to ts

### CODE REVIEW

File: public/jira/releases/semver-releases.ts [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import semver from "semver";
```
Comment (shoumaw) [inline public/jira/releases/semver-releases.ts:1]: This module is not being used, but migrated it to TS anyways for consistency

File: public/jira/releases/semver-releases.ts [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+}
+
+interface SemverRelease {
+  release: string;
+  shortName: string;
+  version: string | undefined;
+  shortVersion: string | undefined;
+}
+
+function partialReleaseName(release: string) {
```
Comment (DavidNic11) [inline public/jira/releases/semver-releases.ts:15]: just flagging this pr too - we need to add return types if that's the direction we go

File: public/jira/releases/unique-trailing-names.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
+interface CharacterMapNode {
```
Comment (DavidNic11) [inline public/jira/releases/unique-trailing-names.ts:1]: can you add a comment for what this module is for (same with the other files that don't have them)

File: public/jira/releases/derive.ts [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+export type DerivedRelease = NormalizedRelease & {
+  names: {
+    semver: boolean;
+    shortName: string;
+    shortVersion: string | null;
+    version: string | null;
+  };
+};
+
+function partialReleaseName(release: string) {
```
[the remark this answers is not in the document]
Comment (shoumaw) [reply public/jira/releases/derive.ts:14]: Sure

**Pull request merged**